staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -720,6 +733,137 @@ func ValidateCustomResourceColumnDefinition(col *apiextensions.CustomResourceCol
 	return allErrs
 }
 
+// validateFieldPath validates a field path against the given schema
+func validateFieldPath(fieldPath string, pathOfFieldPath *field.Path, schema *apiextensions.JSONSchemaProps) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(fieldPath) == 0 {
+		allErrs = append(allErrs, field.Invalid(pathOfFieldPath, fieldPath, "must not be empty"))
+		return allErrs
+	}
+
+	nonexistentFieldError := field.Invalid(pathOfFieldPath, fieldPath, "does not refer to an existent field")
+
+	var s scanner.Scanner
+	s.Init(strings.NewReader(fieldPath))
+	s.Filename = pathOfFieldPath.String()
+	s.Mode = scanner.ScanInts | scanner.ScanIdents | scanner.ScanChars | scanner.ScanStrings
+	s.Error = func(s *scanner.Scanner, msg string) {}
+
+	found := false
+	for true {
+		tok := s.Scan()
+		if tok == scanner.EOF {
+			found = true
+			break
+		}
+		switch schema.Type {
+		case "object":
+			isMapSyntax := false
+			if s.TokenText() == "[" {
+				isMapSyntax = true
+			} else if s.TokenText() != "." {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
+
+			tok = s.Scan()
+			if tok == scanner.EOF {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
+			fieldName := s.TokenText()
+			if isMapSyntax {
+				if tok == scanner.String && len(fieldName) >= 2 && fieldName[0] == '"' &&
+					fieldName[len(fieldName)-1] == '"' {
+					fieldName = fieldName[1 : len(fieldName)-1]
+				} else {
+					allErrs = append(allErrs, nonexistentFieldError)
+					return allErrs
+				}
+			} else if tok != scanner.Ident {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
+
+			if schema.Properties != nil {
+				propertySchema, ok := schema.Properties[fieldName]
+				if ok {
+					schema = &propertySchema
+				} else {
+					allErrs = append(allErrs, nonexistentFieldError)
+					return allErrs
+				}
+			} else if schema.AdditionalProperties != nil {
+				additionalProperty, ok := schema.AdditionalProperties.Schema.Properties[fieldName]
+				if ok {
+					schema = &additionalProperty
+				} else {
+					allErrs = append(allErrs, nonexistentFieldError)
+					return allErrs
+				}
+			} else {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
+
+			if isMapSyntax {
+				if tok == scanner.EOF {
+					allErrs = append(allErrs, nonexistentFieldError)
+					return allErrs
+				}
+				tok = s.Scan()
+				if s.TokenText() != "]" {
+					allErrs = append(allErrs, nonexistentFieldError)
+					return allErrs
+				}
+			}
+		case "array":
+			if s.TokenText() != "[" {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
+			tok = s.Scan()
+			if tok == scanner.EOF {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
+			value, err := strconv.Atoi(s.TokenText())
+			if err != nil {
+				allErrs = append(allErrs, nonexistentFieldError)
+				return allErrs
+			}
```

Comment:
this is expecting the hard-coded field path to be a specific integer index of the array? how would that work correctly? I couldn't say the field path would be `myItems[0]` if I'm writing a dynamic validator that could fail on myItems[1] or myItems[2], right?

Reply:
Have updated to skip numeric index and updated the relevant doc.